src/knw/KNWRobot.h

```diff
@@ -204,126 +17,1140 @@
+         *
+         * @code
+         * // Assuming an IR sensor is wired and connected to digital pin 10
+         * int IRSensorId = 10;
+         * bool success = myRobot->setupIR(IRSensorId, 10);
+         * if (success) {
+         *   // Now ready to use the IR sensor with pingSensorId
+         * }
+         * @endcode
+         */
+        bool setupIR(int id, int pin);
+
+        /**
+         * This uses the IR sensor with the given identifier to scan for beacons that
+         * may be nearby.
+         *
+         * This function uses the IR sensor to scan for IR signals coming from the beacons.
+         * The values that it reads are stored in an internal character buffer, which you
+         * can access using getIR() after running this function.
+         *
+         * <b>Note</b>: every time you want to get fresh values from getIR(), you have to
+         * run this function first.
+         *
+         * @param id The identifier that was provided as the first argument to setupIR().
+         * @return int The number of characters that were read in this given scan.
+         *
+         * Example code:
+         *
+         * @code
+         * // Assuming you have already run the code in setupIR()
+         * int charactersReadFromIR = myRobot->scanIR(IRSensorId);
+         * myRobot->printLCD("Chars read: ");
+         * myRobot->printLCD(charactersReadFromIR);
+         *
+         * // Now print the reading on the next line of the LCD
+         * char* IRCharacters = myRobot->getIR();
+         * myRobot->moveCursor(0, 1);
+         * myRobot->printLCD(IRCharacters);
+         * @endcode
+         */
+        int scanIR(int id);
+
+        /**
+         * This provides the characters that were reading from the most recent call to scanIR().
+         *
+         * This function should be used in conjunction with scanIR(). Every time you want a fresh
+         * IR reading, call scanIR() first. Then, call this function, storing the result in a char*
+         * variable. If you don't call scanIR() before calling this, the internal character buffer
+         * will not refresh and this will continuously provide the same value (or an empty value
+         * if you never ran scanIR() in the first place).
+         *
+         * @return char* A character buffer containing the characters read from the most recent
+         * call to scanIR()
+         *
+         * Example code:
+         *
+         * @code
+         * // Assuming you have already run the code in setupIR()
+         * int charactersReadFromIR = myRobot->scanIR(IRSensorId);
+         * myRobot->printLCD("Chars read: ");
+         * myRobot->printLCD(charactersReadFromIR);
+         *
+         * // Now print the reading on the next line of the LCD
+         * char* IRCharacters = myRobot->getIR();
+         * myRobot->moveCursor(0, 1);
+         * myRobot->printLCD(IRCharacters);
+         * @endcode
+         */
+        unsigned char* getIR();
+
+ private:

```

Comment:
The comments in the public interface are pretty outstanding.  I think the comments explaining the private members could be more robust for maintainability's sake.